### PR TITLE
feat: support instrumentation of gomodule/redigo

### PR DESCRIPTION
**Pull request by RomainMuller**
Uses a new drop-in replacement primitive, which allows the modified call to stand-in as the original one in stack traces (not showing as generated, which could be confusing).

JIRA: APPSEC-53280

### CODE REVIEW

File: internal/injector/aspect/advice/call.go [hunk 1]
```diff
@@ -120,6 +121,53 @@ func (a *appendArgs) AddedImports() []string {
 	return imports
 }
 
+type replaceFunction struct {
+	path string
+	name string
+}
+
+// ReplaceFunction replaces the called function with the provided drop-in replacement. The signature
+// must be compatible with the original function (it may accept a new variadic argument).
```
Comment (eliottness) [inline internal/injector/aspect/advice/call.go:130]: Is there a way to check if it truely is compatible ? Maybe not at runtime but when generating go code from yaml

File: .vscode/launch.json
Comment (eliottness) [inline .vscode/launch.json:1]: I am positive this should not be in this PR

File: internal/injector/aspect/advice/call.go (hunk 1 above)
Comment (RomainMuller) [reply internal/injector/aspect/advice/call.go:130]: The issue is that'd need us to correlate the advice with the join-point, which would be non-trivial...

File: internal/injector/aspect/advice/call.go [hunk 2]
```diff
@@ -138,4 +186,23 @@ func init() {
...
 	}
+	unmarshalers["replace-function"] = func(node *yaml.Node) (Advice, error) {
+		var fqn string
+		if err := node.Decode(&fqn); err != nil {
+			return nil, err
+		}
+
+		var (
+			path string
+			name string
```
Comment (Hellzy) [inline internal/injector/aspect/advice/call.go:197]: That's peanuts but you could declare those together with `fqn`

**Review by Hellzy — APPROVED** [review]

**Pull request merged**